src/main/java/Duke/Command/Parser.java

```diff
@@ -0,0 +1,269 @@
+
+import duke.excpetions.DukeException;
+import duke.task.TaskList;
+
+
+
+public class Parser {
+    private String message;
+
+    /**
+     * @param message Message users take in to be parsed.
+     */
+    public Parser(String message) {
+        this.message = message;
+    }
+
+
+    private static boolean isValidDate(int day, int month, int year, int hour, int minute) {
+        if (((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) && month == 2) {
+            if (day > 29 || day <= 0) {
+                return false;
+            }
+        } else if (month == 2) {
+            if (day > 28 || day <= 0) {
+                return false;
+            }
+        }
+
+        if (month <= 0 || month > 12) {
+            return false;
+        }
+
+        if (month == 1 || month == 3 || month == 5 || month == 7 || month == 8 || month == 10 || month == 12) {
+            if (day > 31 || day <= 0) {
+                return false;
+            }
+        } else if (month != 2) {
+            if (day > 30 || day <= 0) {
+                return false;
+            }
+        }
+
+        if (hour > 24 || hour < 0) {
+            return false;
+        }
+
+        if (minute > 60 || minute < 0) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns a LocalDateTime type that encapsulates the year, month, day, hour, minute of a time input.
+     * The method takes in a String of time and convert into LocalDateTime type.
+     * The format can only be: 1. dd/mm/yyyy hhmm
+     * 2. yyyy-mm-dd
+     *
+     * @param time Time User takes in to be parsed.
+     * @return Parsed time converted in the type of LocalDateTime
+     */
+    public LocalDateTime parseTime(String time) {
+        LocalDateTime parsedTime;
+        int day;
+        int month;
+        int year;
+        int hour;
+        int minute;
+
+        if (time.contains("/") && time.indexOf("/", 3) != -1 && time.contains(" ") && !time.contains("-")) {
+            int endIndex1 = time.indexOf("/");
+            int endIndex2 = time.lastIndexOf(" ");
+            day = Integer.parseInt(time.substring(0, endIndex1));
+
+            Integer dayInteger = day;
+            int endIndex3 = time.indexOf("/", dayInteger.toString().length() + 1);
+            month = Integer.parseInt(time.substring(endIndex1 + 1, endIndex3));
+            year = Integer.parseInt(time.substring(endIndex3 + 1, endIndex2));
+
+            hour = Integer.parseInt(time.substring(endIndex2 + 1).substring(0, 2));
+            minute = Integer.parseInt(time.substring(endIndex2 + 1).substring(2));
+        } else if (time.contains("-")) {
+            try {
+                parsedTime = LocalDate.parse(time).atTime(0, 0);
+                return parsedTime;
+            } catch (DateTimeParseException e) {
+                return null;
+            }
+        } else {
+            return null;
+        }
+
+        //Some Other cases;
```

Comment:
Perhaps best to leave out the semicolons in case it gets confused as comment out code